File: flux-api/src/main/java/com/todo/flux/module/card/service/CardService.java
package com.todo.flux.module.card.service;

import com.todo.flux.module.card.dto.CardAssignRequest;
import com.todo.flux.module.card.dto.CardCreateRequest;
import com.todo.flux.module.card.dto.CardResponse;
import com.todo.flux.module.card.dto.CardUpdateRequest;
import com.todo.flux.module.card.entity.Card;
import com.todo.flux.module.shared.CrudService;

import java.util.List;
import java.util.UUID;

public interface CardService extends CrudService<
                                            Card,
                                            CardCreateRequest,
                                            CardUpdateRequest,
                                            CardResponse,
                                            UUID
                                            > {

    List<CardResponse> listByBoard(UUID boardId);

    CardResponse create(UUID boardId, CardCreateRequest dto);
    CardResponse assignCard(UUID cardId, CardAssignRequest dto);
}

File: flux-frontend/src/app/core/service/card.service.ts
import {Injectable} from '@angular/core';
import {HttpClient} from '@angular/common/http';
import {Observable} from 'rxjs';

import {
    CardCreateRequest,
    CardUpdateRequest,
    CardResponse, CardAssignRequest
} from '../models/card';
import {environment} from '../../environments/environment';

@Injectable({providedIn: 'root'})
export class CardService {
    private baseUrl = `${environment.api}`;

    constructor(private http: HttpClient) {
    }

    getAllByBoard(boardId: string): Observable<CardResponse[]> {
        return this.http.get<CardResponse[]>(`${this.baseUrl}/boards/${boardId}/cards`);
    }

    create(boardId: string, payload: CardCreateRequest): Observable<CardResponse> {
        return this.http.post<CardResponse>(`${this.baseUrl}/boards/${boardId}/cards`, payload);
    }

    getById(cardId: string): Observable<CardResponse> {
        return this.http.get<CardResponse>(`${this.baseUrl}/cards/${cardId}`);
    }

    update(cardId: string, payload: CardUpdateRequest): Observable<CardResponse> {
        return this.http.patch<CardResponse>(`${this.baseUrl}/cards/${cardId}`, payload);
    }

    delete(cardId: string): Observable<void> {
        return this.http.delete<void>(`${this.baseUrl}/cards/${cardId}`);
    }

    assign(cardId: string, payload: CardAssignRequest): Observable<CardResponse> {
        return this.http.patch<CardResponse>(`${this.baseUrl}/cards/${cardId}/assign`, payload);
    }
}
